internal/common/baseclient/baseclient_iigo.go

```diff
@@ -30,17 +30,17 @@ func (c *BaseClient) RuleProposal() string {
 
 // GetClientPresidentPointer is called by IIGO to get the client's implementation of the President Role
 func (c *BaseClient) GetClientPresidentPointer() roles.President {
-	return nil
+	return &BasePresident{}
 }
 
 // GetClientJudgePointer is called by IIGO to get the client's implementation of the Judge Role
 func (c *BaseClient) GetClientJudgePointer() roles.Judge {
-	return nil
+	return &BaseJudge{}
 }
 
 // GetClientSpeakerPointer is called by IIGO to get the client's implementation of the Speaker Role
 func (c *BaseClient) GetClientSpeakerPointer() roles.Speaker {
-	return nil
+	return &BaseSpeaker{}
 }
```

Comment:
Looks like this fixes the issue, however there should be a nil check in IIGO to prevent custom clients from giving a nil pointer to an official, you could either say that implies they're "unfit to rule" i.e. they don't know how to handle the role, or just panic with an explanation so we can figure it out if it happens again.

Reply:
Ok, thats a good plan, implementing now